src/paint.rs

```diff
@@ -17,6 +21,40 @@ pub enum PaintFallback {
     ///
     /// [`<color>`]: https://www.w3.org/TR/css-color-3/
     Color(Color),
+    /// A `var()` function fallback.
+    /// 
+    /// See [`VariableFunction`] for details.
+    Variable(&'a str, Option<Box<PaintFallback<'a>>>),
+}
+
+impl<'a> PaintFallback<'a> {
+    /// Parses a `PaintFallback` from a string.
+    ///
+    /// We can't use the `FromStr` trait because it requires
+    /// an owned value as a return type.
+    #[allow(clippy::should_implement_trait)]
+    fn from_str(text: &'a str) -> Result<Self, Error> {
+        let text = text.trim();
+        match text {
+            "" => return Ok(PaintFallback::Empty),
+            "none" => return Ok(PaintFallback::None),
+            "currentColor" =>return Ok( PaintFallback::CurrentColor),
+            _ => {}
+        }
+
+        let mut s = Stream::from(text);
+        if s.starts_with(b"var(") {
+            let (variable, fallback) = s.parse_var_func()?;
+            let fallback = match fallback {
+                None => None,
+                Some("") => Some(Box::new(PaintFallback::Empty)),
+                Some(other) => Some(Box::new(PaintFallback::from_str(other)?))
```

Comment:
These can be nested arbitrarily, so it also causes an issue where parsing a deeply nested chain of fallbacks will cause stack overflow.